File: C++/BankCartC++/PaymentFactory.cpp
#include "stdafx.h"
#include <iostream>
#include "BankAccount.cpp"

using namespace std;

class PaymentFactory
{
public:
	virtual BankAccount Bank()=0;
};

class BankAccountFactory : public PaymentFactory
{
public:
	BankAccount PaymentFactory::Bank()
	{
		return BankAccount();
	}
};

File: C#/BankC#/PaymentFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CartbankC
{
    abstract class PaymentFactory
    {
        public abstract IPayment Bank();
    }

    class BankAccountFactory : PaymentFactory
    {
        public override IPayment Bank()
        {
            return new BankAccount();
        }
    }
}
